### PR TITLE
[MODULAR GALORE] Makes empty synth bodies printable on the exofab

### PR DESCRIPTION
**Pull request by Majkl-J**
## About The Pull Request

Part 4 of my synthcentric update series

See name

## Why It's Good For The Game

Getting gibbed as synth means you are basically RR currently. Maaybe you could get borged, but that's about it. So I fixed that, and you can now print the synth bodies at robotics

## Changelog

:cl:
add: Synths can now be printed in exofabs
/:cl:

### CODE REVIEW

**Comment by nevimer** [comment]
good idea. we should be able to print human torsos too.

**Comment by Majkl-J** [comment]
> good idea. we should be able to print human torsos too.

You can. It's called humanising a monkey

**Review by StrangeWeirdKitten — APPROVED** [review]

**Comment by nevimer** [comment]
> > good idea. we should be able to print human torsos too.
> 
> You can. It's called humanising a monkey

thats sketchy though, and it leaves people with odd bugs as they remain subtypes of monkeys even when "fixed"

**Comment by StrangeWeirdKitten** [comment]
> > > good idea. we should be able to print human torsos too.
> > 
> > 
> > You can. It's called humanising a monkey
> 
> thats sketchy though, and it leaves people with odd bugs as they remain subtypes of monkeys even when "fixed"

the SAD has always fixed this *decently enough*. But I feel like this is out of scope of this PR. There needs to be a way to print synth torsos because the SAD doesn't fix this.

File: modular_zubbers/code/modules/synths/code/synth_prefab.dm [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/mob/living/carbon/human/species/synth
+	race = /datum/species/synthetic
+
+// Used for printing dead synth bodies
+/mob/living/carbon/human/species/synth/empty
+
+/mob/living/carbon/human/species/synth/empty/Initialize(mapload)
+	. = ..()
+	var/old_death_sound
+	death_sound = null // We don't need them screaming
```
Comment (nevimer) [inline modular_zubbers/code/modules/synths/code/synth_prefab.dm:10]: reset this to initial after

File: modular_zubbers/code/modules/synths/code/synth_prefab.dm [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	death()
+
+	for(var/obj/item/organ/internal/brain/B in organs)
```
Comment (nevimer) [inline modular_zubbers/code/modules/synths/code/synth_prefab.dm:13]: should do var = locate (type) in organs instead

File: modular_zubbers/code/modules/synths/code/synth_prefab.dm [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/mob/living/carbon/human/species/synth
+	race = /datum/species/synthetic
+
+// Used for printing dead synth bodies
+/mob/living/carbon/human/species/synth/empty
+
+/mob/living/carbon/human/species/synth/empty/Initialize(mapload)
+	. = ..()
+	var/old_death_sound
```
Comment (nevimer) [inline modular_zubbers/code/modules/synths/code/synth_prefab.dm:9]: this is never set to anything other than null

File: modular_zubbers/code/modules/synths/code/synth_prefab.dm [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	death_sound = null // We don't need them screaming
+	death()
+
+	for(var/obj/item/organ/internal/brain/B in organs)
+		B.Remove(src)
+		QDEL_NULL(B)
+
+	death_sound = old_death_sound // Now they can die loudly again
```
Comment (nevimer) [inline modular_zubbers/code/modules/synths/code/synth_prefab.dm:17]:
```suggestion
	death_sound = initial(death_sound) // Now they can die loudly again
```

**Comment by nevimer** [comment]
also make sure this costs more than just fixing the body. fixing the body should be much cheaper resource wise.

File: modular_zubbers/code/modules/synths/code/synth_prefab.dm [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+/mob/living/carbon/human/species/synth/empty/Initialize(mapload)
+	. = ..()
+	var/old_death_sound
+	death_sound = null // We don't need them screaming
+	death()
+
+	for(var/obj/item/organ/internal/brain/B in organs)
+		B.Remove(src)
+		QDEL_NULL(B)
```
Comment (nevimer) [inline modular_zubbers/code/modules/synths/code/synth_prefab.dm:15]:
```suggestion
	var/obj/item/organ/internal/brain/B = get_organ_slot(ORGAN_SLOT_BRAIN) 
	if(B)
		B.Remove(src)
		QDEL_NULL(B)
```

**Comment by StrangeWeirdKitten** [comment]
> Member

It doesn't cost anything now to revive a synth / fix the body except time, wires, and welding fuel.

File: modular_zubbers/code/modules/synths/code/synth_prefab.dm (hunk 1 above)
Comment (Majkl-J) [reply modular_zubbers/code/modules/synths/code/synth_prefab.dm:10]: The initial is `null` as death_sound is set by the species post compilation

File: modular_zubbers/code/modules/synths/code/synth_prefab.dm (hunk 4 above)
Comment (Majkl-J) [reply modular_zubbers/code/modules/synths/code/synth_prefab.dm:17]: This is also null

File: modular_zubbers/code/modules/synths/code/synth_prefab.dm (hunk 2 above)
Comment (Majkl-J) [reply modular_zubbers/code/modules/synths/code/synth_prefab.dm:13]: wilco

File: modular_zubbers/code/modules/synths/code/synth_prefab.dm (hunk 3 above)
Comment (Majkl-J) [reply modular_zubbers/code/modules/synths/code/synth_prefab.dm:9]: Ok this was actually missed as I was making this PR on 3 different codebases with IPCs. Woops

**Comment by Majkl-J** [comment]
Tested again after the minor changes and works fine

**Comment by StrangeWeirdKitten** [comment]
Looks fine. No runtimes when I ran it on a test prod of the server. I'm merging it

**Pull request merged**